app/src/main/java/com/asma/tasky/feature_authentication/domain/util/Validation.kt

```diff
@@ -0,0 +1,50 @@
+package com.asma.tasky.feature_authentication.domain.util
+
+import android.util.Patterns
+import com.asma.tasky.core.util.Constants
+
+
+object Validation {
+
+    fun validateName(name: String): AuthError? {
+        val trimmedUsername = name.trim()
+        if (trimmedUsername.isBlank()) {
+            return AuthError.FieldEmpty
+        }
+        if (trimmedUsername.length < Constants.MIN_NAME_LENGTH) {
+            return AuthError.InputTooShort
+        }
+        if (trimmedUsername.length > Constants.MAX_NAME_LENGTH) {
+            return AuthError.InputTooLong
+        }
+        return null
+    }
+
+    fun validateEmail(email: String): AuthError? {
+        val trimmedEmail = email.trim()
+
+        if (!Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
+            return AuthError.InvalidEmail
+        }
```

Comment:
Here, you're using an Android-specific dependency (as you can see in the imports) which doesn't really belong into domain. Instead, I'd recommend to create an abstraction for this to keep this class locally unit testable.

Every time, you have implementation details in domain and want to get rid of that, the procedure is the same:

1. You create an abstraction (interface) in domain
2. You write an implementation of that in data (in this case using Patterns.EMAIL_ADDRESS...)
3. You pass the abstraction to the constructor of your class needing it in domain
4. Now you can decide which implementation you pass when you create that class/use case
For example:

*In domain*
```
interface EmailPatternValidator {
    fun isValidEmailPattern(email: String): Boolean
}
```

*In data*
```
class EmailPatternValidatorImpl: EmailPatternValidator {

    override fun isValidEmailPattern(email: String): Boolean {
        return Patterns.EMAIL_ADDRESS.matcher(email).matches()
    }
}
```

Then inject an `EmailPatternValidator` into your class. Here, you'd therefore need to make `Validation` a class.

Also, is there any reason you use use cases in your app but didn't make `ValidateEmail`, `ValidatePassword` and `ValidateFullName` separate use cases?